pkg/cmd/cli/cmd/create_route.go

```diff
@@ -0,0 +1,454 @@
+  $ %[1]s create route edge my-route --expose frontend
+
+  # Create an edge route that exposes the frontend service and specify a path.
+  # If the route name is omitted, the service name will be re-used.
+  $ %[1]s create route edge --expose frontend --path /assets`
+)
+
+// NewCmdCreateEdgeRoute is a macro command to create an edge route.
+func NewCmdCreateEdgeRoute(fullName string, f *clientcmd.Factory, out io.Writer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "edge [NAME] --expose=SERVICE",
+		Short:   "Create a route that uses edge TLS termination",
+		Long:    edgeRouteLong,
+		Example: fmt.Sprintf(edgeRouteExample, fullName),
+		Run: func(cmd *cobra.Command, args []string) {
+			err := CreateEdgeRoute(f, out, cmd, args)
+			kcmdutil.CheckErr(err)
+		},
+	}
+
+	kcmdutil.AddValidateFlags(cmd)
+	kcmdutil.AddOutputFlagsForMutation(cmd)
+	cmd.Flags().String("hostname", "", "Set a hostname for the new route")
+	cmd.Flags().String("port", "", "Name of the service port or number of the container port the route will route traffic to")
+	cmd.Flags().String("expose", "", "Name of the service that the new route is exposing")
+	cmd.MarkFlagRequired("expose")
+	cmd.Flags().String("path", "", "Path that the router watches to route traffic to the service.")
+	cmd.Flags().String("cert", "", "Path to a certificate file.")
+	cmd.Flags().String("key", "", "Path to a key file.")
+	cmd.Flags().String("ca-cert", "", "Path to a CA certificate file.")
+
+	return cmd
+}
+
+// CreateEdgeRoute implements the behavior to run the create edge route command.
+func CreateEdgeRoute(f *clientcmd.Factory, out io.Writer, cmd *cobra.Command, args []string) error {
+	oc, kc, err := f.Clients()
+	if err != nil {
+		return err
+	}
+	ns, _, err := f.DefaultNamespace()
+	if err != nil {
+		return err
+	}
+	serviceName, err := resolveServiceName(f, kcmdutil.GetFlagString(cmd, "expose"))
+	if err != nil {
+		return err
+	}
+	routeName, err := resolveRouteName(args)
+	if err != nil {
+		return err
+	}
+	route, err := unsecuredRoute(kc, ns, routeName, serviceName, kcmdutil.GetFlagString(cmd, "port"))
+	if err != nil {
+		return err
+	}
+
+	route.Spec.Host = kcmdutil.GetFlagString(cmd, "hostname")
+	route.Spec.Path = kcmdutil.GetFlagString(cmd, "path")
+
+	route.Spec.TLS = new(api.TLSConfig)
+	route.Spec.TLS.Termination = api.TLSTerminationEdge
+	cert, err := fileutil.LoadData(kcmdutil.GetFlagString(cmd, "cert"))
+	if err != nil {
+		return err
+	}
+	route.Spec.TLS.Certificate = string(cert)
+	key, err := fileutil.LoadData(kcmdutil.GetFlagString(cmd, "key"))
+	if err != nil {
+		return err
+	}
+	route.Spec.TLS.Key = string(key)
+	caCert, err := fileutil.LoadData(kcmdutil.GetFlagString(cmd, "ca-cert"))
+	if err != nil {
+		return err
+	}
+	route.Spec.TLS.CACertificate = string(caCert)
+
+	route, err = oc.Routes(ns).Create(route)
+	if err != nil {
+		return err
+	}
+	mapper, typer := f.Object()
```

Comment:
From here down this method is icky.  We should make this easier to use, but I don't see it as a need for this pull.